docs/index.md

```diff
@@ -80,11 +111,11 @@ provider GitHub [discussion board](https://github.com/spectrocloud/terraform-pro
 
 ### Optional
 
-- `api_key` (String, Sensitive)
-- `host` (String)
-- `ignore_insecure_tls_error` (Boolean)
-- `password` (String, Sensitive)
-- `project_name` (String)
-- `retry_attempts` (Number)
-- `trace` (Boolean)
-- `username` (String)
+- `api_key` (String, Sensitive) The Spectro Cloud API key. Can also be set with the `SPECTROCLOUD_API_KEY` environment variable.
+- `host` (String) The Spectro Cloud API host url. Can also be set with the `SPECTROCLOUD_HOST` environment variable. Defaults to https://api.spectrocloud.com
+- `ignore_insecure_tls_error` (Boolean) Ignore insecure TLS error. Defaults to false.
+- `password` (String, Sensitive) The Spectro Cloud user password. Can also be set with the `SPECTROCLOUD_PASSWORD` environment variable.
```

Comment:
I think we need to mention that api_key takes precedence over password and username.

Reply:
@nikchern That's a good point. I can add it to the authentication section of the provider. I think we should keep the descriptions as concise as possible.

Reply:
Here is what I am thinking.
![CleanShot 2022-11-18 at 15 49 26](https://user-images.githubusercontent.com/29551334/202815982-61fe4330-a530-4b2f-b1bc-b35193904031.png)